iFixit/react-commerce
# Tests: Update Jest setup and add snapshot test

## Summary of changes
1. This pull updates the existing `jest.config.js` and install jest `devDependencies` for jest tests.
2. Adds a snapshot test for `ProductListItem`.

## CR Notes
I'm not really familiar with babel config files therefore would like review on https://github.com/iFixit/react-commerce/commit/0e609146e4ededfec40bd12d72d7c945dcd3074f to make sure it doesn't break anything. 

## QA Notes
1. Please make sure the deployed app works properly.
2. It would awesome if you run the jest test locally by running `pnpm install:all`, `cd frontend` and `pnpm test` and make sure it works.

Closes: https://github.com/iFixit/react-commerce/issues/124 and https://github.com/iFixit/ifixit/issues/43590
